### PR TITLE
Added Arguments for Facter module

### PR DESCRIPTION
**Pull request by gamethis**
##### SUMMARY
Added option for Arguments on Facter.
This allows for options to be specified to limit or update the way facter runs

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
facter

##### ADDITIONAL INFORMATION
facter can be a useful tool to verify that puppet runs are actually successful.
this allows one to confirm that puppet run had a desired impact on the system as follows:
```
- name: Run puppet first time with tags
  puppet:
    noop: false
    tags:
      - puppet_enterprise
- name: execute facter
  facter:
    arguments:
      - -p
      - arbitrary_fact
- name: Assert fact from facter
  assert:
    that: puppet_facter.arbitrary_fact == arbitrary_fact_vaule
    fail_msg: "{{ inventory_hostname }} did not register with puppet"
```
this also allows for access to facters other options.

### CODE REVIEW

File: plugins/modules/system/facter.py
Comment (Andersson007) [inline plugins/modules/system/facter.py:24]:
```suggestion
```
This is a default

File: plugins/modules/system/facter.py
Comment (Andersson007) [inline plugins/modules/system/facter.py:41]:
```suggestion
- name: Execute facter no arguments
  community.general.facter:

- name: Execute facter with arguments
  community.general.facter:
```

File: plugins/modules/system/facter.py
Comment (Andersson007) [inline plugins/modules/system/facter.py:17]:
```suggestion
    - Runs the I(facter) discovery program
      (U(https://github.com/puppetlabs/facter)) on the remote system, returning
      JSON data that can be useful for inventory purposes.
```
BTW, the current format is ok as well

**Review by Andersson007 — COMMENTED** [review]
@gamethis thanks for the feature.
Also please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

**Comment by gamethis** [comment]
@Andersson007 thanks for the review.  Sorry for the delay... I have accepted your updates... and added the change log.  Please let me know if we need anything else.

thanks

File: changelogs/fragments/768-facter.yml
Comment (felixfontein) [inline changelogs/fragments/768-facter.yml:2]:
```suggestion
  - facter - added option for ``arguments`` (https://github.com/ansible-collections/community.general/pull/768).
```

File: plugins/modules/system/facter.py
Comment (felixfontein) [reply plugins/modules/system/facter.py:17]: Shouldn't it be `C(facter)`, not `I(facter)`?

File: plugins/modules/system/facter.py
Comment (felixfontein) [inline plugins/modules/system/facter.py:66]:
```suggestion
            cmd += [argument]
```
There's no reason why you want to mangle user input.

Also, why not just do:
```.py
if module.params['arguments']:
    cmd += module.params['arguments']
```
?

Comment (gamethis) [reply plugins/modules/system/facter.py]: switched it up to simpler format

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
If nobody complains, I'll merge this at the beginning of next week.

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@gamethis after this is merged, would you like to be added as a maintainer of the module to .github/BOTMETA.yml?
Your approval (`shipit`, `LGTM`, ..) will be counted by the bot for automerge (needs two) and you'll get notifications about related issues / PRs to your mail (i don't think there will be a lot and nothing obligatory of course). What do you think?

**Comment by gamethis** [comment]
yeah would be happy to be added.

**Comment by felixfontein** [comment]
@gamethis thanks for implementing this!
@Andersson007 thanks for reviewing!

**Pull request merged**
